src/main/java/org/generationcp/middleware/service/pedigree/PedigreeServiceImpl.java

```diff
@@ -147,7 +150,20 @@ public String getCropName() {
 			monitor.stop();
 		}
 	}
-	
+
+	@Override
+	public Map<Integer, String> getCrossExpansionsBulk(final Set<Integer> gids, final Integer level,
+		final CrossExpansionProperties crossExpansionProperties) {
+
+		final Iterable<List<Integer>> partition = Iterables.partition(gids, MAX_GID_LIST_SIZE);
+		final Map<Integer, String> crossExpansions = new HashMap<>();
+
+		for (final List<Integer> partitionedGidList : partition) {
+			crossExpansions.putAll(this.getCrossExpansions(new HashSet<>(partitionedGidList), level, crossExpansionProperties));
+		}
+		return crossExpansions;
+	}
+
```

Comment:
@IntegratedBreedingPlatform/developers 
I've copied this common workaround directly to the pedigree service, so that we don't have to replicate every time we need it. Let me know if you have any concern with this.